api.ts

```diff
@@ -2184,13 +2258,30 @@ export function makeSyncTable<
   }
 
   const responseHandler = generateObjectResponseHandler({schema: formulaSchema});
-  const execute = async function exec(params: ParamValues<ParamDefsT>, context: SyncExecutionContext) {
-    const {result, continuation} = (await wrappedExecute(params, context)) || {};
+  const execute = async function exec(
+    params: ParamValues<ParamDefsT>,
+    context: SyncExecutionContext,
+  ): Promise<InternalSyncFormulaResult<K, L, SchemaT>> {
+    const syncResult = (await wrappedExecute(params, context)) || {};
     const appliedSchema = context.sync.schema;
+    const result = responseHandler({body: syncResult.result || [], status: 200, headers: {}}, appliedSchema) as Array<
+      ObjectSchemaDefinitionType<K, L, SchemaT>
+    >;
+
+    if (isSyncFormulaPartialResult(syncResult)) {
+      return {
+        result,
+        continuation: syncResult.continuation,
+      };
+    }
     return {
-      result: responseHandler({body: result || [], status: 200, headers: {}}, appliedSchema),
-      continuation,
-    } as SyncFormulaResult<K, L, SchemaT>;
+      result,
+      syncExecutionCompletionMetadata: {
+        // It's ok if this is empty
+        ...syncResult.syncExecutionCompletionMetadata,
+        syncExecutionCompletionMetadataVersion: CurrentSyncExecutionCompletionMetadataVersion,
```

Comment:
I guess maybe this is ok, I guess I had only been thinking about persisting the completion data on the `coda` side and wanting those payloads to be versioned, as an implementation detail. Versioning the SDK output might be useful but feels maybe orthogonal?

Reply:
Seems like CurrentSyncExecutionCompletionMetadataVersion in the metadata is kind of implied by the SDK version

Do we expect them to diverge and be able to change independently?

Reply:
I'll just delete all this stuff. Y'all are right, the versioning can all be done on the Coda side.